test/integration/backward_compatible/map/MapDistinctAggregatorTest.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const { should } = require('chai');
+should();
+const RC = require('../../RC');
+const long = require('long');
+const {
+    Aggregators,
+    Client,
+    Predicates
+} = require('../../../../');
+const TestUtil = require('../../../TestUtil');
+
+describe('MapDistinctAggregatorTest', function () {
+
+    let cluster;
+    let client;
+    let map;
+
+    const mixedMapValues = ['0', '1', 1.1, long.fromNumber(1), {}, { a: 1 }, true, true, false, '0'];
+
+    const fixedMapValues = ['0', '1', '1', '2', 'asd', 'asd', '1231231asd'];
+
+    before(async function () {
+        cluster = await RC.createCluster(null, null);
+        await RC.startMember(cluster.id);
+        client = await Client.newHazelcastClient({ clusterName: cluster.id });
+
+        // distinct aggregator is added in 5.0 to the Node.js client
+        TestUtil.markClientVersionAtLeast(this, '5.0');
+        // distinct aggregator fixed for non-java clients in 4.0 https://github.com/hazelcast/hazelcast/pull/15506
+        TestUtil.markServerVersionAtLeast(this, client, '4.0');
```

Comment:
We don't need this check as the 4.0+ clients are only compatible with 4.0 members. Hence, the member version cannot be less than that